components/sso-saml/org.wso2.carbon.identity.sso.saml.ui/src/main/resources/web/sso-saml/add_service_provider_finish.jsp

```diff
@@ -126,88 +126,70 @@
         }
 
         if (Boolean.parseBoolean(request.getParameter(SAMLSSOUIConstants.ENABLE_ATTRIBUTE_PROFILE))) {
-
             String claimsCountParameter = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyCounter");
             if (claimsCountParameter != null && !"".equals(claimsCountParameter)) {
-                try {
-                    int claimsCount = Integer.parseInt(claimsCountParameter);
-                    for (int i = 0; i < claimsCount; i++) {
-                        String claim = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyName" + i);
-                        if (claim != null && !"".equals(claim) && !"null".equals(claim)) {
-                            String[] currentClaims = serviceProviderDTO.getRequestedClaims();
-                            boolean isClaimAlreadyAdded = false;
-                            for (String currentClaim : currentClaims) {
-                                if (claim.equals(currentClaim)) {
-                                    isClaimAlreadyAdded = true;
-                                    break;
-                                }
-                            }
-
-                            if (!isClaimAlreadyAdded) {
-                                serviceProviderDTO.addRequestedClaims(claim);
+                int claimsCount = Integer.parseInt(claimsCountParameter);
+                for (int i = 0; i < claimsCount; i++) {
+                    String claim = SAMLSSOUIUtil.getSafeInput(request, "claimPropertyName" + i);
+                    if (claim != null && !"".equals(claim) && !"null".equals(claim)) {
+                        String[] currentClaims = serviceProviderDTO.getRequestedClaims();
+                        boolean isClaimAlreadyAdded = false;
+                        for (String currentClaim : currentClaims) {
```

Comment:
Doesn't this particular code block indicate the need of "currentClaims" being a map? AFAIU, being able to look-up for claims is required in order to determine if a given claim is already added or not. Much efficient option, IMO, would be to populate all claims in a map and later use that to look up existing claims. WDYT?